src/anomalib/data/utils/path.py

```diff
@@ -58,13 +58,15 @@ def _prepare_files_labels(
     if isinstance(extensions, str):
         extensions = (extensions,)
 
+    assert all(extension.startswith(".") for extension in extensions), "All extension names must start with the dot."
```

Comment:
@abc-125, would this assert statement be helpful to the user as well? If yes, would it be an idea to replace the assert statement with the `raise`?